File: redsheet-portal/src/app/database-models/v-our-batna.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vOurBatna {
    constructor(vOurBatna?: vOurBatna )  {
        let properties = [ "ourbatnaId", "ourbatnaValue", "projectRequirementId",];
        for (let property of properties)
            if (vOurBatna && vOurBatna[property])
                this[property] = vOurBatna[property];
    }
 
	ourbatnaId : number =   0 ;
 
    @required()
    @maxLength(1000)
	ourbatnaValue : string =   undefined;
 
    @range(1,2147483647)
	projectRequirementId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vOurbatna.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vOurBatnas", Schema = "dbo")]
	[Serializable]
	public partial class vOurBatna  
	{
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int OurbatnaId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OurbatnaValue { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectRequirementId { get; set; } 

	public vOurBatna()
	{
    }

    }
}

